src/ai/index.js

```diff
@@ -0,0 +1,550 @@
+            // conversation[0] = { role: "user", content: `${question1}` };
+        }
+        // todo Steve: include extra information here to provide to ai
+        let extra = {
+            worldObjectId: realityEditor.worldObjects.getBestWorldObject().objectId,
+        }
+        console.log(conversation);
+        realityEditor.network.postQuestionToAI(conversation, extra);
+    }
+    
+    function getAnswer(category, answer) {
+        console.log(`%c This question is of category ${category}`, 'color: blue');
+        
+        // todo Steve new: preprocess the answer in map.js, and then send out the processed answer to the dialogue
+        //  but since the processed answer got fed straight into the ai prompt, not sure if this is a good idea, or the names need to be converted to ids again, and fed back to the ai
+        let html = map.postprocess(answer);
+        pushAIDialogue(html);
+    }
+    
+    function getToolAnswer(category, tools) {
+        console.log(`%c This question is of category ${category}`, 'color: blue');
+        
+        let result = tools.split('\n');
+        console.log(result);
+        let bestWorldObject = realityEditor.worldObjects.getBestWorldObject();
+        let frames = [];
+        for (let frame of Object.values(bestWorldObject.frames)) {
+            if (!result.includes(frame.src)) continue;
+            frames.push(frame);
+        }
+        pushToolDialogue(frames, result);
+    }
+    
+    function showDialogue() {
+        aiContainer.style.animation = `slideToRight 0.2s ease-in forwards`;
+    }
+    
+    function hideDialogue() {
+        aiContainer.style.animation = `slideToLeft 0.2s ease-in forwards`;
+    }
+    
+    function hideEndpointApiKeyAndShowSearchTextArea() {
+        endpointArea.style.display = 'none';
+        apiKeyArea.style.display = 'none';
+        searchTextArea.style.display = 'block';
+        adjustTextAreaSize();
+    }
+
+    function setupEventListeners() {
+        endpointArea.addEventListener('keydown', (e) => {
+            e.stopPropagation();
+            if (e.key === 'Enter') {
+                e.preventDefault();
+                if (endpointArea.value === '' || apiKeyArea.value === '') return;
+                realityEditor.network.postAiApiKeys(endpointArea.value, apiKeyArea.value, true);
+            }
+        })
+        apiKeyArea.addEventListener('keydown', (e) => {
+            e.stopPropagation();
+            if (e.key === 'Enter') {
+                e.preventDefault();
+                if (endpointArea.value === '' || apiKeyArea.value === '') return;
+                realityEditor.network.postAiApiKeys(endpointArea.value, apiKeyArea.value, true);
+            }
+        })
+        
+        searchTextArea.addEventListener('input', function() {
+            // adjustTextAreaSize();
+        });
+        
+        // todo Steve: below 4 doesn't work
+        searchTextArea.addEventListener('mousedown', (e) => {e.stopPropagation();});
```

Comment:
I believe the reason these aren't working is that they should be pointerdown pointerup pointermove etc.

Reply:
Make sense!